### PR TITLE
docs: add glossary for common terms

### PR DESCRIPTION
**Pull request by bilalshaikh42**

### CODE REVIEW

**Comment by jonrkarr** [comment]
Embedding the entire glossary in every page seems a bit repetitive. Would a more prominent top-level menu item be sufficient?

File: docs/snippets/glossary.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+*[SED-ML]: Simulation Experiment Description Markup Language
```
Comment (jonrkarr) [inline docs/snippets/glossary.md:1]: I think it would be helpful to sort this in alphabetical order

File: docs/index.md [hunk 2]
```diff
@@ -198,3 +198,4 @@ The BioSimulators platform supports a broad range of modeling frameworks, model
 </a>
 </div>
 </div>
+--8<-- "glossary.md"
```
Comment (jonrkarr) [inline docs/index.md:201]: Rather than embedding into every page, the index could be added to the menu via `.pages`

**Review by jonrkarr — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 2 above)
Comment (bilalshaikh42) [reply docs/index.md:201]: > 
The glossary isn't actually displayed but provides tooltips over the words defined across the page. 
This page has a example of this extension https://squidfunk.github.io/mkdocs-material/reference/abbreviations/#configuration

**Pull request merged**